Neos.ContentRepository/Classes/Eel/FlowQueryOperations/CanEvaluateNodeContextTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Eel\FlowQueryOperations;
+
+use Neos\ContentRepository\Domain\Projection\Content\NodeInterface;
+use Neos\ContentRepository\Domain\Projection\Content\TraversableNodeInterface;
+
+trait CanEvaluateNodeContextTrait
+{
+    /**
+     * @param $context
+     * @return bool
+     */
+    public function checkContextForNodeInterface($context): bool
+    {
+        if (is_array($context)) {
+            return count($context) > 0 ? reset($context) instanceof NodeInterface : true;
+        } elseif ($context instanceof \Traversable) {
+            foreach ($context as $item) {
+                return $item instanceof NodeInterface;
```

Comment:
This would return true if the first array item is a node, but the second is not.

Reply:
yes but this was done previously like this as well. Its just super simple duck-typing to speed up performance ...